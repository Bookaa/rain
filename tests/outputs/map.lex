keyword 'import'
name 'array'
newline
keyword 'import'
name 'iter'
newline
keyword 'let'
name 'double'
symbol '='
keyword 'func'
symbol '('
name 'n'
symbol ')'
operator '->'
name 'n'
operator '*'
int 2
newline
keyword 'let'
name 'main'
symbol '='
keyword 'func'
symbol '('
symbol ')'
indent
keyword 'let'
name 'a'
symbol '='
'table'
operator '::'
name 'array'
newline
name 'a'
symbol '['
int 0
symbol ']'
symbol '='
int 10
newline
name 'a'
symbol '['
int 1
symbol ']'
symbol '='
int 11
newline
name 'a'
symbol '['
int 2
symbol ']'
symbol '='
int 12
newline
name 'a'
symbol '['
int 3
symbol ']'
symbol '='
int 13
newline
name 'print'
symbol '('
string 'keys from a:'
symbol ')'
newline
keyword 'for'
name 'x'
keyword 'in'
name 'a'
symbol ':'
name 'keys'
symbol '('
symbol ')'
indent
name 'print'
symbol '('
name 'x'
symbol ')'
newline
dedent
newline
name 'print'
symbol '('
string 'values from a:'
symbol ')'
newline
keyword 'for'
name 'x'
keyword 'in'
name 'a'
symbol ':'
name 'values'
symbol '('
symbol ')'
indent
name 'print'
symbol '('
name 'x'
symbol ')'
newline
dedent
newline
name 'print'
symbol '('
string 'doubled values from a:'
symbol ')'
newline
keyword 'for'
name 'x'
keyword 'in'
name 'iter'
symbol '.'
name 'map'
symbol '('
name 'a'
symbol ':'
name 'values'
symbol '('
symbol ')'
symbol ','
name 'double'
symbol ')'
indent
name 'print'
symbol '('
name 'x'
symbol ')'
newline
dedent
newline
keyword 'let'
name 'b'
symbol '='
name 'a'
symbol ':'
name 'map'
symbol '('
name 'double'
symbol ')'
newline
name 'print'
symbol '('
string 'keys from b:'
symbol ')'
newline
keyword 'for'
name 'x'
keyword 'in'
name 'b'
symbol ':'
name 'keys'
symbol '('
symbol ')'
indent
name 'print'
symbol '('
name 'x'
symbol ')'
newline
dedent
newline
name 'print'
symbol '('
string 'values from b:'
symbol ')'
newline
keyword 'for'
name 'x'
keyword 'in'
name 'b'
symbol ':'
name 'values'
symbol '('
symbol ')'
indent
name 'print'
symbol '('
name 'x'
symbol ')'
newline
dedent
newline
dedent
newline
EOF
