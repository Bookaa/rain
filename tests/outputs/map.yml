!program;1
stmts:
- !import;1
  name: array
  rename: null
- !import;1
  name: iter
  rename: null
- !assn;2
  export: false
  let: true
  lhs: !name;1
    value: double
  rhs: !func;1
    body: !return;1
      value: !binary;1
        lhs: !name;1
          value: n
        op: '*'
        rhs: !int;1
          value: 2
    params:
    - n
- !assn;2
  export: false
  let: true
  lhs: !name;1
    value: main
  rhs: !func;1
    body: !block;2
      expr: null
      stmts:
      - !assn;2
        export: false
        let: true
        lhs: !name;1
          value: a
        rhs: !table;1
          parent: !name;1
            value: array
      - !assn;2
        export: false
        let: false
        lhs: !index;1
          lhs: !name;1
            value: a
          rhs: !int;1
            value: 0
        rhs: !int;1
          value: 10
      - !assn;2
        export: false
        let: false
        lhs: !index;1
          lhs: !name;1
            value: a
          rhs: !int;1
            value: 1
        rhs: !int;1
          value: 11
      - !assn;2
        export: false
        let: false
        lhs: !index;1
          lhs: !name;1
            value: a
          rhs: !int;1
            value: 2
        rhs: !int;1
          value: 12
      - !assn;2
        export: false
        let: false
        lhs: !index;1
          lhs: !name;1
            value: a
          rhs: !int;1
            value: 3
        rhs: !int;1
          value: 13
      - !call;1
        args:
        - !str;1
          value: 'keys from a:'
        catch: false
        func: !name;1
          value: print
      - !for;1
        body: !block;2
          expr: null
          stmts:
          - !call;1
            args:
            - !name;1
              value: x
            catch: false
            func: !name;1
              value: print
        func: !method;1
          args: []
          catch: false
          lhs: !name;1
            value: a
          rhs: !str;1
            value: keys
        name: x
      - !call;1
        args:
        - !str;1
          value: 'values from a:'
        catch: false
        func: !name;1
          value: print
      - !for;1
        body: !block;2
          expr: null
          stmts:
          - !call;1
            args:
            - !name;1
              value: x
            catch: false
            func: !name;1
              value: print
        func: !method;1
          args: []
          catch: false
          lhs: !name;1
            value: a
          rhs: !str;1
            value: values
        name: x
      - !call;1
        args:
        - !str;1
          value: 'doubled values from a:'
        catch: false
        func: !name;1
          value: print
      - !for;1
        body: !block;2
          expr: null
          stmts:
          - !call;1
            args:
            - !name;1
              value: x
            catch: false
            func: !name;1
              value: print
        func: !call;1
          args:
          - !method;1
            args: []
            catch: false
            lhs: !name;1
              value: a
            rhs: !str;1
              value: values
          - !name;1
            value: double
          catch: false
          func: !index;1
            lhs: !name;1
              value: iter
            rhs: !str;1
              value: map
        name: x
      - !assn;2
        export: false
        let: true
        lhs: !name;1
          value: b
        rhs: !method;1
          args:
          - !name;1
            value: double
          catch: false
          lhs: !name;1
            value: a
          rhs: !str;1
            value: map
      - !call;1
        args:
        - !str;1
          value: 'keys from b:'
        catch: false
        func: !name;1
          value: print
      - !for;1
        body: !block;2
          expr: null
          stmts:
          - !call;1
            args:
            - !name;1
              value: x
            catch: false
            func: !name;1
              value: print
        func: !method;1
          args: []
          catch: false
          lhs: !name;1
            value: b
          rhs: !str;1
            value: keys
        name: x
      - !call;1
        args:
        - !str;1
          value: 'values from b:'
        catch: false
        func: !name;1
          value: print
      - !for;1
        body: !block;2
          expr: null
          stmts:
          - !call;1
            args:
            - !name;1
              value: x
            catch: false
            func: !name;1
              value: print
        func: !method;1
          args: []
          catch: false
          lhs: !name;1
            value: b
          rhs: !str;1
            value: values
        name: x
    params: []
