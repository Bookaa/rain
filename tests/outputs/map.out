keys from a:
0
1
2
3
values from a:
10
11
12
13
doubled values from a:
20
22
24
26
key/values from b:
0 = 20
1 = 22
2 = 24
3 = 26
